src/content/ui/interactivity/focus.md

```diff
@@ -3,179 +3,222 @@ title: Understanding Flutter's keyboard focus system
-If you would like more control over where the focus goes when you remove it from
-a node, explicitly focus another node instead of calling `unfocus()`, or use the
-focus traversal mechanism to find another node with the `focusInDirection`,
+**Do create `FocusNode` and `FocusScopeNode` objects in a stateful widget.**
+: When you finish using `FocusNode` and `FocusScopeNode`,
+  you need to dispose of them.
+  For this reason, only create them inside of a stateful widget's `State` object.
+  You can override the stateful widget's `dispose` method to dispose of the
+  focus objects.
+
+**Do set the `debugLabel` of a focus node widget.**
+: This helps with diagnosing focus issues.
+
+**Do call the `requestFocus()` method on a node to request the primary focus.**
+: Call this method from an node that has passed a node
+  it owns to a descendant where you want to the focus.
+
+**Do use `focusNode.requestFocus()`.**
+: You don't need to call `FocusScope.of(context).requestFocus(focusNode)`.
+  The `focusNode.requestFocus()` method works the same with better performance.
+
+**Don't allocate a new `FocusNode` object for each build.**
+: This can cause memory leaks.
+  It might also cause a loss of focus when the widget rebuilds while
+  the node has focus.
+
+**Don't use the same `FocusNode` for multiple widgets.**
+: If you do, the widgets fight over managing the attributes of the node.
+  That might result in an unexpected outcome.
+
+**Don't set the `onKeyEvent` callback on a `FocusNode` that `Focus` manages.**
+: To set focus to respond to a key event, use an `onKeyEvent` handler.
+  Surround the desired widget subtree with a new [`Focus`][] widget
+  with its `onKeyEvent` attribute set to your handler.
+  If you don't want the widget to be able to take primary focus,
+  set `canRequestFocus: false` on it.
+  Do this because the `onKeyEvent` attribute on the `Focus` widget
+  can be set to something else in a later build. If that happens,
+  the attribute overwrites the `onKeyEvent` handler you set on the node.
+  This applies to a `FocusScopeNode` that a [`FocusScope`][] widget
+  manages.
+
+### Tell a node to give up focus
+
+The `FocusNode.unfocus()` API tells a node to "give up the focus".
+While it does remove focus from the node,
+note that you can't "unfocus" all nodes.
+If you remove focus from a node, it must pass the focus somewhere else,
+since there is _always_ a primary focus. When a node calls `unfocus()`,
+focus can pass to one of two nodes.
+It can pass to either the nearest `FocusScopeNode`
+or a node in the same scope that had focus earlier.
+Which one depends upon the `disposition` argument you give to `unfocus()`.
+To better control where the focus passes when you remove it from a node,
+you have two options.
+You can choose to set focus to another node or use the focus traversal mechanism.
+To find another node, this mechanism can use the `focusInDirection`,
 `nextFocus`, or `previousFocus` methods on `FocusNode`.
```

Comment:
I think it would be better to talk about the `UnfocusDisposition` options first before this paragraph, since this is really describing an option to using `unfocus`, and the options go with `unfocus`, not this.

Also, you could add explicitly focusing another node with `requestFocus()` to the list of options somewhere, so there are three options, really.

Reply:
All I have to work with is https://api.flutter.dev/flutter/widgets/UnfocusDisposition.html. If there's something else I should consider, I'm happy to take suggestions.